internal/controller/controller.go

```diff
@@ -43,6 +43,30 @@ type Route struct {
 	log logr.Logger
 }
 
+func shouldSync(log logr.Logger, route *routev1.Route) bool {
```

Comment:
**Self-review:** At first, I wanted to remove the logging from this function. But removing logging meant that I would not be able to choose at which level each message would show, and I don't think logging is that bad even though it makes the function do side effects.